Dockerfile

```diff
@@ -40,14 +40,20 @@ RUN set -x; \
     && git submodule update --init skins \
     && git submodule update --init vendor \
     && cd extensions \
-    # VisualEditor
+    # Extensions
     # TODO: make submodules shallow clones?
-    && git submodule update --init VisualEditor \
-    && cd VisualEditor \
-    && git checkout $MEDIAWIKI_VERSION \
-    && git submodule update --init \
-    && cd .. \
-    && git submodule update --init Math
+    && git submodule update --init --recursive VisualEditor \
+    && git submodule update --init --recursive Math \
+    && git submodule update --init --recursive EventBus \
+    && git submodule update --init --recursive Scribunto \
+    && git submodule update --init --recursive ParserFunctions \
+    && git submodule update --init --recursive SyntaxHighlight_GeSHi \
+    && git submodule update --init --recursive Cite \
+    && git submodule update --init --recursive Echo \
+    && git submodule update --init --recursive Flow \
+    && git submodule update --init --recursive PageImages \
+    && git submodule update --init --recursive TextExtracts
```

Comment:
Let me add mobileview and citoid in a follow-up.

As far as I know, submodules are versioned in each release branch, so initializing them should check out the right version. This is incidentally also making it impossible to check out those submodules in a shallow fashion, as the submodule pointer might not be pointing to HEAD of master.